include/onnxruntime/core/framework/allocator.h

```diff
@@ -45,9 +45,11 @@ constexpr const char* HIP_PINNED = "HipPinned";
 constexpr const char* OpenVINO_CPU = "OpenVINO_CPU";
 constexpr const char* OpenVINO_GPU = "OpenVINO_GPU";
 
-
 constexpr size_t kAllocAlignment = 256;
 
+class IAllocator;
+void* AllocateBufferWithOptions(std::shared_ptr<IAllocator>& allocator, size_t size, bool use_reserve, Stream* stream, WaitNotificationFn wait_fn);
+
 // forward declaration
 class SessionState;
```

Comment:
May no longer be needed

Reply:
Unfortunately i still need it. The reason is i don't want to expose the BFCArena API to the shared execution provider interface, as it will expose too many symbols. so i need this function to hide the underline symbols.

Reply:
I meant, there is not any more references to SessionState on this header.

Reply:
ah ok, i misunderstood it. yes u r right.